google/resource_dataproc_job_test.go

```diff
@@ -0,0 +1,595 @@
+					resource.TestCheckResourceAttrSet("google_dataproc_job.sparksql", "status.0.state"),
+					resource.TestCheckResourceAttrSet("google_dataproc_job.sparksql", "status.0.state_start_time"),
+
+					// Unique job config
+					testAccCheckDataprocJobAttrMatch(
+						"google_dataproc_job.sparksql", "sparksql_config", &job),
+
+					// Wait until job completes successfully
+					testAccCheckDataprocJobCompletesSuccessfully("google_dataproc_job.sparksql", &job),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckDataprocJobDestroy(s *terraform.State) error {
+	config := testAccProvider.Meta().(*Config)
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "google_dataproc_job" {
+			continue
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("Unable to verify delete of dataproc job ID is empty")
+		}
+		attributes := rs.Primary.Attributes
+
+		_, err := config.clientDataproc.Projects.Regions.Jobs.Get(
+			config.Project, attributes["region"], rs.Primary.ID).Do()
+
+		if err != nil {
+			if gerr, ok := err.(*googleapi.Error); ok && gerr.Code == 404 {
+				return nil
+			} else if ok {
+				return fmt.Errorf("Error make GCP platform call: http code error : %d, http message error: %s", gerr.Code, gerr.Message)
+			}
+			return fmt.Errorf("Error make GCP platform call: %s", err.Error())
+		}
+		return fmt.Errorf("Dataproc job still exists")
+	}
+
+	return nil
+}
+
+func testAccCheckDataprocJobCompletesSuccessfully(n string, job *dataproc.Job) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		config := testAccProvider.Meta().(*Config)
+
+		attributes := s.RootModule().Resources[n].Primary.Attributes
+		region := attributes["region"]
+		project, err := getTestProject(s.RootModule().Resources[n].Primary, config)
+		if err != nil {
+			return err
+		}
+
+		jobCompleteTimeoutMins := 3
+		waitErr := dataprocJobOperationWait(config, region, project, job.Reference.JobId,
+			"Awaiting Dataproc job completion of failure", jobCompleteTimeoutMins, 1)
+		if waitErr != nil {
+			return waitErr
+		}
+
+		completeJob, err := config.clientDataproc.Projects.Regions.Jobs.Get(
+			config.Project, region, job.Reference.JobId).Do()
+		if err != nil {
+			return err
+		}
+		if completeJob.Status.State != "DONE" {
+			return fmt.Errorf("Job did not complete successfully, instead status: %s", completeJob.Status.State)
+		}
+
+		return nil
+	}
+}
+
+func testAccCheckDataprocJobExists(n string, job *dataproc.Job) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Terraform resource Not found: %s", n)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No ID is set for Dataproc job")
+		}
+
+		config := testAccProvider.Meta().(*Config)
+		jobId := s.RootModule().Resources[n].Primary.ID
+		found, err := config.clientDataproc.Projects.Regions.Jobs.Get(
+			config.Project, rs.Primary.Attributes["region"], jobId).Do()
```

Comment:
any reason to use `config.Project` here but `getTestProject` above? I'm fine with either one, but may as well be consistent